src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> lastNode = last;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        last = newNode;
 
+        if (lastNode == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkElementIndex(index);
+
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentElement = findElement(index);
+            Node<T> prevLink = currentElement.prev;
+            Node<T> newNode = new Node<>(prevLink, value, currentElement);
+            currentElement.prev = newNode;
 
+            if (prevLink == null) {
+                first = newNode;
+            } else {
+                prevLink.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (T t: list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index);
+        return findElement(index).item;
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkPositionIndex(index);
+        findElement(index).item = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndex(index);
+        Node<T> removedElement = findElement(index);
+        Node<T> prevLink = removedElement.prev;
+        Node<T> nextLink = removedElement.next;
+
+        if (prevLink == null) {
+            first = nextLink;
+        } else {
+            prevLink.next = nextLink;
+            removedElement.prev = null;
+        }
+
+        if (nextLink == null) {
+            last = prevLink;
+        } else {
+            nextLink.prev = prevLink;
+            removedElement.next = null;
+        }
+
+        T removedItem = removedElement.item;
+        removedElement.item = null;
+        size--;
+
+        return removedItem;
     }
 
     @Override
     public T remove(T t) {
+        for (int i = 0; i < size; i++) {
+            if (t == findElement(i) || t.equals(findElement(i).item)) {
```

Comment:
```suggestion
            if (t == findElement(i).item || t.equals(findElement(i).item)) {
```

Reply:
better create separate variable for `findElement(i).item`

Reply:
okey, fixed)